src/app/GitCommands/Git/GitModule.cs

```diff
@@ -2850,7 +2850,8 @@ public void UnstageFileToRemove(string file)
             _gitExecutable.RunCommand(args);
         }
 
-        /// <summary>Dirty but fast. This sometimes fails.</summary>
+        /// <summary>Attempt to read the branch name from the HEAD file instead of calling a git command.</summary>
+        /// <remarks>Dirty but fast. This sometimes fails. In reftable repos, it always returns ".invalid".</remarks>
         public static string GetSelectedBranchFast(string? repositoryPath, bool emptyIfDetached = false)
         {
             if (string.IsNullOrEmpty(repositoryPath))
```

Comment:
Maybe in a followup
reftable is set in the config:
https://git-scm.com/docs/reftable#_repository_format

So in that case the file should never be read at all.